src/watchmaker/workers/salt.py

```diff
@@ -360,6 +381,97 @@ def _get_failed_states(self, state_ret):
             failed_states = state_ret
         return failed_states
 
+    def _install_pip(self):
+        if not self.py_exec:
+            raise WatchmakerException("Python executable not set")
+
+        get_pip = os.path.join(
+            os.path.abspath(
+                os.path.join(os.path.dirname(__file__), '..', '..')),
+            'vendor',
+            'pypa',
+            'get-pip',
+            '2.6',
+            'get-pip.py')
+        self.log.info(
+            'Attempting pip install using get-pip (%s)...', get_pip)
+        cmd = [
+            self.py_exec,
+            get_pip,
+            "--index-url",
+            self.pip_index
+        ]
+        self.call_process(cmd)
+
+    def _upgrade_pip(self):
+        if not self.py_exec:
+            raise WatchmakerException("Python executable not set")
+
+        self.log.info('Attempting to upgrade pip...')
+        cmd = [
+            self.py_exec,
+            "-m",
+            "pip",
+            "install",
+            "--upgrade",
+            "pip",
+            "--index-url",
+            self.pip_index
+        ]
+        try:
+            self.call_process(cmd)
```

Comment:
We can just use the `raise_error` argument to disable the exception...
```suggestion
            self.call_process(cmd, raise_error=False)
```